elements/lisk-p2p/test/functional/actions/send_to_peer.ts

```diff
@@ -42,28 +31,31 @@ describe('P2P.sendToPeer', () => {
 	it('should send message to a specific peer within the network', async () => {
 		const firstP2PNode = p2pNodeList[0];
 
-		const targetPeer = firstP2PNode.getConnectedPeers()[1];
+		const targetPeer = firstP2PNode.getConnectedPeers()[0];
 
 		firstP2PNode.sendToPeer(
 			{
 				event: 'foo',
 				data: 123,
 			},
-			`${targetPeer.ipAddress}:${targetPeer.wsPort}`,
+			constructPeerId(targetPeer.ipAddress, targetPeer.wsPort),
 		);
 
-		await wait(300);
-
-		expect(collectedMessages.length).to.equal(1);
-		expect(collectedMessages[0])
-			.to.have.property('nodePort')
-			.which.is.equal(targetPeer.wsPort);
-		expect(collectedMessages[0]).to.have.property('message');
-		expect(collectedMessages[0].message)
-			.to.have.property('event')
-			.which.is.equal('foo');
-		expect(collectedMessages[0].message)
-			.to.have.property('data')
-			.which.is.equal(123);
+		p2pNodeList[1].on('messageReceived', msg => {
```

Comment:
this is a better approach. Just you need to call `done` in the callback because this test would always pass as soon as it's done with the execution of top-level code.